### PR TITLE
Hotfix: fixing issue when lambda arguments causes NPE on Identifier Naming rule (closing #106)

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
1) Fixed NPE for lambda identifiers
2) Added tests
3) Fixed typos in old code and updated readme

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 1]
```diff
@@ -86,40 +88,62 @@ class IdentifierNaming : Rule("identifier-naming") {
...
+     * getting identifiers (aka variable names) from parent nodes like PROPERTY
+     * The trick here is to handle DESTRUCTURING_DECLARATION correctly, as it does not have IDENTIFIER leaf
+     */
+    private fun extractVariableIdentifiers(node: ASTNode): List<ASTNode> {
+        val destructingDeclaration = node.getFirstChildWithType(ElementType.DESTRUCTURING_DECLARATION)
+        val result = if (destructingDeclaration != null) {
+            destructingDeclaration.getAllChildrenWithType(ElementType.DESTRUCTURING_DECLARATION_ENTRY)
+                    .map { it.getIdentifierName()!! }
+        } else {
+            mutableListOf(node.getIdentifierName()!!)
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:142]: Why mutable list here?

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter2/KdocMethodsFixTest.kt [hunk 2]
```diff
@@ -40,3 +35,10 @@ class KdocMethodsFixTest : FixTestBase("test/paragraph2/kdoc/package/src/main/ko
         fixAndCompare("KdocMethodsFullExpected.kt", "KdocMethodsFullTested.kt")
     }
 }
+
+class KdocMethodsFixTest : FixTestBase("test/paragraph2/kdoc/methods", KdocMethods()) {
+    @Test
+    fun `KDoc template should be placed before modifiers`() {
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter2/KdocMethodsFixTest.kt:41]: This test if fixed in master branch, so this fix can be reverted here.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt (hunk 1 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:142]: fixed

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter2/KdocMethodsFixTest.kt (hunk 2 above)
Comment (akuleshov7) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter2/KdocMethodsFixTest.kt]: reverted

**Pull request merged**